crates/iota-core/src/authority/shared_object_congestion_tracker.rs

```diff
@@ -1581,4 +1632,213 @@ mod object_cost_tests {
             .is_none()
         );
     }
+
+    #[rstest]
+
+    fn test_try_schedule_allow_overshoot(
+        #[values(
+            PerObjectCongestionControlMode::TotalGasBudget,
+            PerObjectCongestionControlMode::TotalTxCount
+        )]
+        mode: PerObjectCongestionControlMode,
+        #[values(true, false)] assign_min_free_execution_slot: bool,
+    ) {
+        let shared_obj_0 = ObjectID::random();
+        let shared_obj_1 = ObjectID::random();
+
+        let tx_gas_budget = 100;
+
+        // instantiate the tracker with some initial debts.
+        let shared_object_congestion_tracker = match mode {
+            PerObjectCongestionControlMode::None => unreachable!(),
+            PerObjectCongestionControlMode::TotalGasBudget => {
+                // Construct object execution cost as following
+                //                199   301
+                // object 0:            |
+                // object 1:      |
+                //
+                // burst limit is 100 + 200 = 300
+                // tx cost is 100 (gas budget)
+                SharedObjectCongestionTracker::new_for_test(
+                    [(shared_obj_0, 301), (shared_obj_1, 199)],
+                    mode,
+                    assign_min_free_execution_slot,
+                )
+            }
+            PerObjectCongestionControlMode::TotalTxCount => {
+                // Construct object execution cost as following
+                //                3     4
+                // object 0:            |
+                // object 1:      |
+                //
+                // overall limit is 2 + 2 = 4
+                // tx cost is 1 (tx count)
+                SharedObjectCongestionTracker::new_for_test(
+                    [(shared_obj_0, 4), (shared_obj_1, 3)],
+                    mode,
+                    assign_min_free_execution_slot,
+                )
+            }
+        };
+
+        // Set max_execution_duration_per_commit to allow 1 transaction to
+        // go through before an overshoot occurs.
+        let max_execution_duration_per_commit = match mode {
+            PerObjectCongestionControlMode::None => unreachable!(),
+            PerObjectCongestionControlMode::TotalGasBudget => tx_gas_budget,
+            PerObjectCongestionControlMode::TotalTxCount => 2,
```

Comment:
```suggestion
            PerObjectCongestionControlMode::TotalTxCount => 1,
```

Why is it 2 if the comment says "to allow 1 transaction to go through"?

Reply:
Made the comments clearer now. You have to take initial object debts into account as well as the limit.

Reply:
Thanks for making it clearer.